packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -140,14 +237,6 @@ export class TagManager {
       include.indexOf(this.resourceTypeName) === -1) {
       return false;
     }
-    if (this.tags[key]) {
-      if (this.tags[key].props.priority !== undefined) {
-        return priority >= this.tags[key].props.priority!;
-      }
-    }
-    if (this.removedTags[key]) {
-      return priority >= this.removedTags[key];
-    }
-    return true;
+    return priority >= (this.priorities.get(key) || 0);
```

Comment:
Is this simplification enough? or did you want me to look at removing the include/exclude checks above as well.